great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -342,25 +348,52 @@ def _generate_selectable_from_schema_name_and_table_name(
             schema=schema_name,
         )
 
+    @overload
     def _generate_selectable_from_query(
         self,
         query: str,
         dialect: GXSqlDialect,
+        create_temp_table: Literal[True],
+        temp_table_schema_name: Optional[str] = ...,
+    ) -> sqlalchemy.Table:
+        ...
+
+    @overload
+    def _generate_selectable_from_query(
+        self,
+        query: Selectable,
+        dialect: GXSqlDialect | None,
+        create_temp_table: Literal[False],
+        temp_table_schema_name: Optional[str] = ...,
+    ) -> sqlalchemy.TextClause:
+        ...
+
+    def _generate_selectable_from_query(
+        self,
+        query: str | Selectable,
+        dialect: GXSqlDialect | None,
         create_temp_table: bool,
         temp_table_schema_name: Optional[str] = None,
-    ) -> sqlalchemy.Table:
+    ) -> sqlalchemy.Table | sqlalchemy.TextClause | Selectable:
         """Helper method to generate Selectable from query string.
 
         Args:
             query (str): query passed in as RuntimeBatchRequest.
             dialect (GXSqlDialect): Needed for _create_temporary_table, since different backends name temp_tables differently.
-            create_temp_table (bool): Should we create a temp_table?
+            create_temp_table (bool): Should we create a temp_table? If not a `TextClause` will be returned instead of a Table.
             temp_table_schema_name (Optional[str], optional): Optional string for temp_table schema.  Defaults to None.
 
         Returns:
-            sqlalchemy.Table: SqlAlchemy Table that is Selectable.
+            sqlalchemy.Table: SqlAlchemy Table that is Selectable or a TextClause.
         """
         if not create_temp_table:
+            if isinstance(query, Selectable):
+                compiled_query = query.compile(
+                    dialect=self.sql_engine_dialect,
+                    compile_kwargs={"literal_binds": True},
+                )
+                text_query = str(compiled_query)
+                return sa.text(text_query)
```

Comment:
I think this should be done in `get_batch_data_and_markers()` before creating the `SqlAlchemyBatchData()` object.

Reply:
Did it